### PR TITLE
Add tests for command globbing

### PR DESCRIPTION
**Pull request by TravisEz13**
# PR Summary

Add tests for command globbing
  - Also, add `-noprofile` to places where `pwsh` is launched so they don't fail locally.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 1]
```diff
@@ -245,6 +245,48 @@ Describe "TabCompletion" -Tags CI {
...
+                    command = "'.\[ra]emove-powershell.ps1'"
+                    expectedCommand = "'.${separator}remove-powershell.ps1'"
+                    name = "'.${separator}[ra]emove-powershell.ps1'"
+                }
+            )
+
+            Push-Location ${TestDrive}\
+        }
+
+        it "Input <name> should successfully complete" -TestCases $gcmWithWildcardCases {
```
Comment (adityapatwardhan) [inline test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: Maybe a better name for `$gcmWithWildcardCases`

**Review by adityapatwardhan — APPROVED** [review]

File: test/powershell/engine/Basic/CommandDiscovery.Tests.ps1 [hunk 2]
```diff
@@ -84,4 +84,33 @@ Describe "Command Discovery tests" -Tags "CI" {
...
+            setup -f '1.ps1' -content "'$secondResult'"
+
+            $gcmWithWildcardCases = @(
+                @{command = '.\?[tb]est1?.ps1'; expectedCommand = '[test1].ps1'; expectedCommandCount =1; name = '''.\?[tb]est1?.ps1'''}
+                @{command = (Join-Path ${TestDrive}  -ChildPath '?[tb]est1?.ps1'); expectedCommand = '[test1].ps1'; expectedCommandCount =1 ; name = '''.\?[tb]est1?.ps1'' by fully qualified path'}
+                @{command = '.\[test1].ps1'; expectedCommand = '1.ps1'; expectedCommandCount =1; name = '''.\[test1].ps1'''}
+                @{command = (Join-Path ${TestDrive}  -ChildPath '[test1].ps1'); expectedCommand = '1.ps1'; expectedCommandCount =1 ; name = '''.\[test1].ps1'' by fully qualified path'}
+            )
+
+            Push-Location ${TestDrive}\
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply test/powershell/engine/Basic/CommandDiscovery.Tests.ps1]: I am testing relative path.  I need to be in that path.  This is the location we change path.

[the remark this answers is not in the document]
Comment (iSazonov) [reply test/powershell/engine/Basic/CommandDiscovery.Tests.ps1]: Ah, it is setup -f. Maybe put the push-location after the setup -f?

File: test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1 [hunk 3]
```diff
@@ -245,6 +245,48 @@ Describe "TabCompletion" -Tags CI {
...
+    Context "Script name completion" {
+        BeforeAll {
+            setup -f 'install-powershell.ps1' -content ""
+            setup -f 'remove-powershell.ps1' -content ""
+            $separator = [system.io.path]::DirectorySeparatorChar
+
+            $scriptWithWildcardCases = @(
+                @{
+                    command = '.\install-*.ps1'
+                    expectedCommand = ".${separator}install-powershell.ps1"
```
[the remark this answers is not in the document]
Comment (TravisEz13) [reply test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1]: updated

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**